components/vc/rear/src/mcManager.c

```diff
@@ -177,7 +177,7 @@ static void mcManager_periodic_100Hz(void)
 
 #if FEATURE_IS_ENABLED(FEATURE_REVERSE)
     CAN_gear_E direction = CAN_GEAR_FORWARD;
-    const bool direction_valid = CANRX_get_signal(VEH, STW_gear, &direction) == CANRX_MESSAGE_VALID;
+    const bool direction_valid = CANRX_get_signal(VEH, VCFRONT_gear, &direction) == CANRX_MESSAGE_VALID;
```

Comment:
Why are you doing all this gear state logic on vcfront only to consume it here? You can avoid a hop if you just do the logic on vc rear, I think. Then the CAN state just becomes for logging, seemingly.